libcxx/test/std/ranges/range.adaptors/range.stride.view/adaptor.pass.cpp

```diff
@@ -0,0 +1,147 @@
+using BidirStrideView        = std::ranges::stride_view<BasicTestView<bidirectional_iterator<int*>>>;
+using RandomAccessStrideView = std::ranges::stride_view<BasicTestView<random_access_iterator<int*>>>;
+
+using SizedForwardStrideView =
+    std::ranges::stride_view<BasicTestView<random_access_iterator<int*>, random_access_iterator<int*>>>;
+using SizedInputStrideView = std::ranges::stride_view<BasicTestView<SizedInputIterator, SizedInputIterator>>;
+
+static_assert(std::ranges::forward_range<ForwardStrideView>);
+static_assert(std::ranges::bidirectional_range<BidirStrideView>);
+static_assert(std::ranges::random_access_range<RandomAccessStrideView>);
+static_assert(std::ranges::forward_range<SizedForwardStrideView>);
+static_assert(std::sized_sentinel_for<std::ranges::iterator_t<SizedForwardStrideView>,
+                                      std::ranges::iterator_t<SizedForwardStrideView>>);
+static_assert(std::sized_sentinel_for<std::ranges::iterator_t<SizedInputStrideView>,
+                                      std::ranges::iterator_t<SizedInputStrideView>>);
+
+constexpr bool test() {
+  constexpr int N = 3;
+  int arr[N]      = {1, 2, 3};
+
+  // Test that `std::views::stride` is a range adaptor.
+  // Check various forms of
+
+  // view | stride
+  // two tests: first with stride of 1; second with stride of 2.
+  {
+    using View                                                          = BasicTestView<cpp17_input_iterator<int*>>;
+    auto view                                                           = make_input_view(arr, arr + N);
+    std::same_as<std::ranges::stride_view<View>> decltype(auto) strided = view | std::views::stride(1);
+    auto strided_iter                                                   = strided.begin();
+
+    // Check that the begin() iter views arr[0]
+    assert(*strided_iter == arr[0]);
+
+    // Check that the strided_iter, after advancing it 2 * 1 steps, views arr[2].
+    std::ranges::advance(strided_iter, 2);
+    assert(*strided_iter == arr[2]);
+  }
+  {
+    using View                                                          = BasicTestView<cpp17_input_iterator<int*>>;
+    auto view                                                           = make_input_view(arr, arr + N);
+    std::same_as<std::ranges::stride_view<View>> decltype(auto) strided = view | std::views::stride(2);
+    auto strided_iter                                                   = strided.begin();
+
+    assert(*strided_iter == arr[0]);
+    std::ranges::advance(strided_iter, 1);
+    assert(*strided_iter == arr[2]);
+  }
+
+  // adaptor | stride
+  // two tests: first with stride of 1; second with stride of 2.
+  const auto i2 = [](int i) { return i * 2; };
+  {
+    auto view                           = make_input_view(arr, arr + N);
+    const auto transform_stride_partial = std::views::transform(i2) | std::views::stride(1);
+
+    auto transform_stride_applied      = transform_stride_partial(view);
```

Comment:
I tend to agree -- long identifiers can make tests harder to read at a glance because you have to parse each long identifier before you understand what the test does -- especially when the long identifiers differ in only in subtle ways.